homeassistant/components/alexa/capabilities.py

```diff
@@ -1179,6 +1183,18 @@ def capability_resources(self):
             self._resource.add_mode(f"{cover.ATTR_POSITION}.custom", ["Custom"])
             return self._resource.serialize_capability_resources()
 
+        if self.instance == f"{light.DOMAIN}.{light.ATTR_EFFECT}":
+            self._resource = AlexaModeResource(["Effect"], False)
+
+            effect_list = self.entity.attributes.get(light.ATTR_EFFECT_LIST, 0)
+            if effect_list:
+                for effect in effect_list:
+                    self._resource.add_mode(
+                        f"{light.ATTR_EFFECT}.{effect}", [f"{effect}"]
+                    )
```

Comment:
Since we're not going to limit the effect_list to the supported effects; sanitize for potential values that have space, period, plus sign, etc. in the effect mode identifier and the friendly name. Make sure you can still interpret the resulting mode identifier back to the integration effect name when Alexa sends a setMode directive.

The WLED integration has a large list of effect names to test with.
```
effect_list: Android,BPM,Blink,Blink Rainbow,Breathe,Chase,Chase,Chase Flash,Chase Flash Rnd,Chase Rainbow,Chase Rainbow,Chase Random,Circus,Colorful,Colorloop,Colortwinkle,Colorwaves,Dark Sparkle,Dissolve,Dissolve Rnd,Dual Scan,Dual Scanner,Dynamic,Fade,Fill Noise,Fire 2012,Fire Flicker,Fireworks,Gradient,Halloween,ICU,In In,In Out,Juggle,Lake,Lighthouse,Lightning,Loading,Mega Strobe,Merry Christmas,Meteor,Multi Comet,Noise 1,Noise 2,Noise 3,Noise 4,Oscillate,Out In,Out Out,Palette,Pride 2015,Railway,Rain,Rainbow,Rainbow Runner,Random Colors,Red & Blue,Ripple,Running,Running 2,Saw,Scan,Scanner,Smooth Meteor,Solid,Sparkle,Sparkle+,Stream,Stream 2,Strobe,Strobe Rainbow,Sweep,Sweep Random,Traffic Light,Tri Chase,Tri Fade,Tri Wipe,Twinkle,Twinklefox,Wipe,Wipe Random
```

Reply:
Great! Thank you will add a test case for those too.

Reply:
This came to mind; we can base64 encode the effect name as the mode value during discovery. Alexa will then send back the base64 encoded value when the effect is chosen. Allowing us to easily translate it back into the actual name for the `light.turn_on` service. 

That way we're just converting for effect names for friendly name utterances: Removing underscores, hyphens, and written name for other symbols.  e.g. `Sparkle+` should become `Sparkle Plus`. 

You said you had success with numbers in the friendly names; but in my experience I've had better experience exposing the written name vs. the number. I'll leave that up to you. 

```json
{
  "configuration": {
    "ordered": false,
    "supportedModes": [
      {
        "value": "effect.U3BhcmtsZSs",
        "modeResources": {
          "friendlyNames": [
            {
              "@type": "text",
              "value": {
                "text": "Sparkle Plus",
                "locale": "en-US"
              }
            }
          ]
        }
      }
    ]
  }
}
```

Reply:
> This came to mind; we can base64 encode the effect name as the mode value during discovery.

Brilliant idea.

> You said you had success with numbers in the friendly names; but in my experience I've had better user experience exposing the written name vs. the number. I'll leave that up to you.

I'd be happy enough to continue with the symbols in place and see what bug reports we get back (if any) and adjust from there. Given Alexa handled the numbers OK i'd imagine it has support for other symbols too.